plugin/keyring_vault/keyring_vault-test/keyring_vault-test.cc

```diff
@@ -0,0 +1,291 @@
+}
+
+int main(int argc, char **argv)
+{
+  my_thread_global_init();
+  mysql_mutex_init(0, &LOCK_verbose, MY_MUTEX_INIT_FAST);
+
+  keyring::system_charset_info= &my_charset_utf8_general_ci;
+  srand(time(NULL));
+  unsigned long long threads_store_number= 0;
+  unsigned long long threads_remove_number= 0;
+  unsigned long long threads_fetch_number= 0;
+  unsigned long long threads_generate_key_number= 0;
+  int number_of_keys_to_store= 0;
+  int number_of_keys_to_fetch= 0;
+  int number_of_keys_to_remove= 0;
+  my_thread_handle *otid;
+  unsigned long long i;
+  void *tret;
+
+  if (argc != 12)
+  {
+    fprintf(stderr, "Usage: keyring_vault_test <path_to_keyring_vault_conf> <threads_store_number> "
+                    "<threads_remove_number> <threads_fetch_number> "
+                    "<threads_generate_key_number> <number_of_keys_to_store> "
+                    "<number_of_keys_to_fetch> <number_of_keys_to_generate> "
+                    "<max_generated_key_length> <random_keys> <verbose>\n");
+    return -1;
+  }
+
+  my_init();
+
+  keyring_vault_config_file = (char*)my_malloc(PSI_NOT_INSTRUMENTED, strlen(argv[1])+1, MYF(0));
+  strcpy(keyring_vault_config_file, argv[1]);
+  threads_store_number= atoll(argv[2]) ;
+  threads_remove_number= atoll(argv[3]) ;
+  threads_fetch_number= atoll(argv[4]);
+  threads_generate_key_number= atoll(argv[5]);
+  number_of_keys_to_store= atoi(argv[6]);
+  number_of_keys_to_fetch= atoi(argv[7]);
+  number_of_keys_to_generate= atoi(argv[8]);
+  max_generated_key_length= atoi(argv[9]);
+  random_keys= atoi(argv[10]);
+  verbose= atoi(argv[11]);
+  number_of_keys_to_remove= number_of_keys_to_store + number_of_keys_to_generate;
+
+  unsigned long long threads_number= threads_store_number + threads_fetch_number + threads_remove_number +
+                                     threads_generate_key_number;
+
+  if (!(otid= (my_thread_handle*) my_malloc(PSI_NOT_INSTRUMENTED,
+       threads_number * sizeof(*otid), MYF(0))))
+    return -2;
+
+  st_plugin_int plugin_info;
+  plugin_info.name.str= const_cast<char*>("keyring_vault");
+  plugin_info.name.length= strlen("keyring_vault");
+
+  if (keyring_vault_init(&plugin_info))
+    return 1;
+
+  for (i= 0; i < threads_store_number; i++)
+    if (mysql_thread_create(PSI_NOT_INSTRUMENTED, &otid[i], NULL, store, (void *) &number_of_keys_to_store))
+      return 2;
+
+  for (i= 0; i < threads_fetch_number; i++)
+    if (mysql_thread_create(PSI_NOT_INSTRUMENTED, &otid[threads_store_number + i], NULL, fetch, (void *) &number_of_keys_to_fetch))
+      return 3;
+
+  for (i= 0; i < threads_remove_number; i++)
+    if (mysql_thread_create(PSI_NOT_INSTRUMENTED, &otid[threads_store_number + threads_fetch_number + i], NULL, remove, (void *) &number_of_keys_to_remove))
+      return 4;
+
+  for (i= 0; i < threads_generate_key_number; i++)
+    if (mysql_thread_create(PSI_NOT_INSTRUMENTED, &otid[threads_store_number + threads_fetch_number + threads_remove_number + i],
```

Comment:
It is just a test program - supposed to be written quickly. This also applies to unit tests, I did not put much effort in making it clean code. Those tests change too rapidly to care about design.